### PR TITLE
Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by zewdu444**
### Blog app: add authorization rules
 
we've accomplished the following with @MarkoKermi in this milestone.

- [x]  install and setup `CanCanCan `
- [x] Add a role column to the users table
- [x] A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). 
- [x] A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). 
- [x] Fix linter error

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -5,6 +5,7 @@ def index
   end
 
   def show
+    @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
   end
```
Comment (TadesseDev) [inline app/controllers/posts_controller.rb:11]: - [x] Good job adding the CanCanCan `ability` class Team. But, as required for the project and to correctly set up the user authorization using [CanCanCan](https://github.com/CanCanCommunity/cancancan), kindly make sure you use CanCanCan for Post authorization in Post Controller. 
> Something like...
```ruby  
class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Correct GitHFlow  ✅
 - [x]  Good commit history  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (zewdu444) [reply app/controllers/posts_controller.rb:11]: @TadesseDev  We appreciate your thoughtful feedback and we have made changes.

**Review by codecaiine — APPROVED** [review]
Hi @zewdu444 @MarkoKermi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good authorization rules implementation ✔️ 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**